services/elasticache/pkg/resource/cache_parameter_group/custom_update_api.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cache_parameter_group
+
+import (
+	"context"
+	ackcompare "github.com/aws/aws-controllers-k8s/pkg/compare"
+	svcapitypes "github.com/aws/aws-controllers-k8s/services/elasticache/apis/v1alpha1"
+)
+
+// Implements specialized logic for update CacheParameterGroup.
+func (rm *resourceManager) customUpdateCacheParameterGroup(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	diffReporter *ackcompare.Reporter,
+) (*resource, error) {
+	desiredParameters := desired.ko.Spec.ParameterNameValues
+	latestParameters := latest.ko.Status.ParameterNameValues
```

Comment:
This should be `latest.ko.Spec.ParameterNameValues`. There's no need to have ParameterNameValues in both the Spec and Status because `latest` is the CR constructed from the ReadOne operation...

Reply:
Removed the field from status